### ISSUE TITLE
Stack bench doesnt work when used with options --work-dir and --stack-yaml 

### ISSUE THREAD

**Issue by wdanilo** [commercialhaskell/stack#1908]
Hello!
My build setup looks in the following way:
```
- build
    - stack-develop.yaml
    - stack-nightly.yaml
    - stack.yaml

- pkg.cabal
- stack.yaml (symlink to build/stack.yaml)
```

When I run compilation using `stack --work-dir .stack-work --stack-yaml build/stack-develop.yaml build --copy-bins --bench --ghc-options="-O -dshow-passes"` everything works and the benchmark files get build. But if I run ``stack --work-dir .stack-work --stack-yaml build/stack-develop.yaml build bench` I get the following output:
`name of package: benchmarks`

If I instead copy the `stack-develop.yaml` config to the root folder and change it according to its location (relative paths have to have one redirection less), everything works like it should work.

**Comment by mgsloan** [comment]
If you run a single benchmark, do you get its output?  It's using the same logic that's used to determine if a build log is outputted - if it's a target and the last thing executing, IIRC.  See this issue: https://github.com/commercialhaskell/stack/issues/426

`stack bench` is a bit of a rougher part of stack.  I've just realized we should probably execute them sequentially: https://github.com/commercialhaskell/stack/issues/1913

This means that they wouldn't run in parallel with other tasks, and so it's reasonable to send their output directly to the user.

**Comment by rrnewton** [comment]
So it sounds like once #1913 is fixed, this will automatically be fixed as a consequence of the existing print-serial-output policy?

**Comment by mgsloan** [comment]
Yes, hopefully so.  It may require some tweaks to that logic, though, remains to be seen.  Good thing to consider when working on  #1913!

**Comment by mgsloan** [comment]
Actually, I was simply wrong about #1913.  So, we are actually executing tests and benchmarks sequentially.  In that case, it makes sense to me to output it to the terminal by default.
